examples/az_cli.py

```diff
@@ -13,3 +13,57 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Demo CLI tool for Azure."""
+
+from typing import TYPE_CHECKING
+
+from libcloudforensics.providers.azure.internal import account
+from libcloudforensics.providers.azure import forensics
+
+if TYPE_CHECKING:
+  import argparse
+
+
+def ListInstances(args: 'argparse.Namespace') -> None:
+  """List instances in Azure subscription.
+
+  Args:
+    args (argparse.Namespace): Arguments from ArgumentParser.
+  """
+
+  az_account = account.AZAccount(args.subscription_id)
+  instances = az_account.ListInstances(
+      resource_group_name=args.resource_group_name)
+
+  print('Instances found:')
+  for instance in instances:
+    boot_disk = instances[instance].GetBootDisk().name
```

Comment:
Also possible to do this:

```suggestion
  for instance in instances.values():
    boot_disk = instance.GetBootDisk().name
```